provider/dyn.go

```diff
@@ -108,11 +108,47 @@ type DynConfig struct {
 	DynVersion    string
 }
 
+// ZoneSnapshot stores a single recordset for a zone for a single serial
+type ZoneSnapshot struct {
+	serials   map[string]int
+	endpoints map[string][]*endpoint.Endpoint
+}
+
+// GetRecordsForSerial retrieves from memory the last known recordset for the (zone, serial) tuple
+func (snap *ZoneSnapshot) GetRecordsForSerial(zone string, serial int) []*endpoint.Endpoint {
+	lastSerial, ok := snap.serials[zone]
+	if !ok {
+		// no mapping
+		return nil
+	}
+
+	if lastSerial != serial {
+		// outdated mapping
+		return nil
+	}
+
+	endpoints, ok := snap.endpoints[zone]
+	if !ok {
+		// probably a bug
+		return nil
+	}
+
+	return endpoints
+}
+
+// StoreRecordsForSerial associates a result set with a (zone, serial)
+func (snap *ZoneSnapshot) StoreRecordsForSerial(zone string, serial int, records []*endpoint.Endpoint) {
+	snap.serials[zone] = serial
```

Comment:
do you also want to check if the zone exists?

Reply:
That's a good suggestion in case many zones are configured but a few have a type

Reply:
Do you want to add it?

Reply:
Yes, I've amended the commit: https://github.com/kubernetes-incubator/external-dns/pull/675/files#diff-5b8f502ee66724d8eaf2173dbbd1ed35R599

There is no elegant way to catch the 404 but to parse the error string.

Reply:
Got it, I didn't saw it.